File: robocode2-server/src/main/java/dev/robocode/tankroyale/server/events/ScannedBotEvent.java
package dev.robocode.tankroyale.server.events;

import lombok.Value;

/**
 * Event sent when a bot got scanned.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class ScannedBotEvent implements Event {

	/** Turn number when event occurred */
	int turnNumber;

	/** Bot id of the bot that scanned the bot */
	int scannedByBotId;

	/** Bot id of the bot that got scanned */
	int scannedBotId;
	
	/** Energy level of the scanned bot */
	double energy;
	
	/** X coordinate of the scanned bot */
	double x;

	/** Y coordinate of the scanned bot */
	double y;

	/** Driving direction of the scanned bot */
	double direction;
	
	/** Speed of the scanned bot */
	double speed;
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/event/ScannedBotEvent.kt
package dev.robocode.tankroyale.server.event

/** Event sent when a bot got scanned. */
class ScannedBotEvent (
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bot id of the bot that scanned the bot */
    val scannedByBotId: Int,

    /** Bot id of the bot that got scanned */
    val scannedBotId: Int,

    /** Energy level of the scanned bot */
    val energy: Double,

    /** X coordinate of the scanned bot */
    val x: Double,

    /** Y coordinate of the scanned bot */
    val y: Double,

    /** Driving direction of the scanned bot */
    val direction: Double,

    /** Speed of the scanned bot */
    val speed: Double,

) :  Event()